### PR TITLE
Game logic Branch : Milestone 3

### PR DESCRIPTION
**Pull request by SmithVyne**
- Separated all the classes
- Removed `puts` from the class' files
- We created methods to handle game properties.
- Fixed all errors
- We created separate files for all classes.
- We created different classes for different game functions.

Please review nicely :)

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,84 +1,69 @@
...
+    end
+  end
 end
-# rubocop:enable Style/StringLiterals, Metrics/LineLength, Style/WordArray, Lint/UnneededDisable, Lint/RedundantCopDisableDirective, Naming/MethodParameterName
+
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
+require_relative '../lib/game.rb'
+
+Game.new.play
```
Comment (kobiyoyo) [inline bin/main.rb:69]: This should be instantiated properly like this 
````ruby
game = Game.new
game.play
````

File: bin/main.rb [hunk 2]
```diff
@@ -1,84 +1,69 @@
...
+        x.any?(element)
+      end
+    end
+  end
 end
-# rubocop:enable Style/StringLiterals, Metrics/LineLength, Style/WordArray, Lint/UnneededDisable, Lint/RedundantCopDisableDirective, Naming/MethodParameterName
+
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
+require_relative '../lib/game.rb'
```
Comment (kobiyoyo) [inline bin/main.rb:67]: This should be at the top of the file

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class Game
+  include Gameprops
+  attr_accessor :game_on, :players_moves, :moves, :filled
```
Comment (kobiyoyo) [inline lib/game.rb:3]: Use attr_readers/attr_writers instead of attr_accessors.

File: lib/player.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Players
+  attr_accessor :players_moves
```
Comment (kobiyoyo) [inline lib/player.rb:2]: Use attr_readers/attr_writers instead of attr_accessors.

**Comment by AhmedTohamy01** [comment]
Hi Desmond, 

Thank you for your comments, the stickler hangup on this request and doesn't want to work, may be because we created the pull request first and then enable the stickler from stickler website.. Any way, that's why we closed this on one and created a new pull request, and the stickler works fine on it .

File: lib/player.rb (hunk 4 above)
Comment (SmithVyne) [reply lib/player.rb:2]: We needed both writers and readers here

**Pull request closed without merging**